maintenance_team_hierarchy/models/maintenance_team.py

```diff
@@ -0,0 +1,67 @@
+# Copyright 2019 Creu Blanca
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class MaintenanceTeam(models.Model):
+
+    _parent_name = "parent_id"
+    _parent_store = True
+    _parent_order = "name"
+    _inherit = "maintenance.team"
+
+    parent_id = fields.Many2one("maintenance.team", ondelete="restrict")
+    parent_path = fields.Char(index=True)
```

Comment:
As https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-16.0 recommends, it should be useful adding this, not tested:
```suggestion
    parent_path = fields.Char(index=True, unaccent=False)
```